store/src/db/sql.rs

```diff
@@ -113,6 +115,35 @@ pub fn select_accounts_by_block_range(
     Ok(result)
 }
 
+/// Select the latest account details by account id from the DB using the given [Connection].
+///
+/// # Returns
+///
+/// The latest account details, or an error.
+pub fn get_account_details(
+    conn: &mut Connection,
+    account_id: AccountId,
+) -> Result<Account> {
+    let mut stmt = conn.prepare(
+        "SELECT nonce, vault, storage, code FROM account_details WHERE account_id = ?1;",
+    )?;
+
+    let mut rows = stmt.query(params![u64_to_value(account_id)])?;
+    let Some(row) = rows.next()? else {
+        return Err(DatabaseError::AccountNotOnChain(account_id));
+    };
```

Comment:
Nit: I think this is a little more readable:

```suggestion
    let row = rows.next()?.ok_or(Err(DatabaseError::AccountNotOnChain(account_id)))?;
```